backend/market/views.py

```diff
@@ -57,167 +68,163 @@ def get_queryset(self):
         return Offer.objects.filter(user=user)
 
 
-class Properties(viewsets.ModelViewSet):
+def apply_filters(
+    queryset, params, filter_mappings, user=None, is_sublet=False, tag_field="tags__name"
+):
+    # Build dynamic filters based on filter mappings
+    filters = {}
+    for param, field in filter_mappings.items():
+        if param_value := params.get(param):
+            filters[field] = param_value
+
+    # Handle seller/owner filtering based on user ownership
+
+    # Apply basic filters to the queryset
+
+    # Exclude specific categories if provided
+    if not is_sublet:
+        queryset = queryset.exclude(category__name__in=["Sublet"])
+        if params.get("seller", "false").lower() == "true" and user:
+            filters["seller"] = user
+        else:
+            filters["expires_at__gte"] = timezone.now()
+    else:
+        queryset = queryset.filter(item__category__name__in=["Sublet"])
+        if params.get("seller", "false").lower() == "true" and user:
+            filters["item__seller"] = user
+        else:
+            filters["item__expires_at__gte"] = timezone.now()
+
+    queryset = queryset.filter(**filters)
+
+    # Apply tag filtering iteratively if "tags" parameter is provided
```

Comment:
why r tags special

Reply:
They all have the same field name (tags__name for items, item__tags__name for sublets)